frontend/src/components/GroupedItems.tsx

```diff
@@ -0,0 +1,85 @@
+import {Item} from "../type/Item.tsx";
+import {useState} from "react";
+
+type ItemListProps = {
+    items: {
+        item: Item,
+        quantity: string
+    }[],
+    checkbox: boolean
+}
+
+function ListItemWithoutCheckbox(props: { item: { name: string; quantity: string } }) {
+    return <li>{props.item.quantity} {props.item.name}</li>;
+}
+
+function ListItemWithCheckbox(props: { item: { name: string; quantity: string } }) {
+    const [isChecked, setIsChecked] = useState(false);
+
+
+    const handleCheckboxChange = () => {
+        setIsChecked(!isChecked);
+    };
+
+    const inputId = `checkbox-${props.item.name}`;
+
+    return <li className="flex items-center">
+                <input
+                    type="checkbox"
+                    id={inputId}
+                    checked={isChecked}
+                    onChange={handleCheckboxChange}
+                    className="h-5 w-5 cursor-pointer rounded border border-gray-300 checked:bg-blue-600 checked:border-blue-600"
+                />
+                <label htmlFor={inputId} className={`ml-2 text-gray-800 ${isChecked ? 'line-through text-gray-500' : ''}`}>
+                    {props.item.quantity} {props.item.name}
+                </label>
+    </li>
+}
+
+export default function GroupedItems({items, checkbox}: ItemListProps) {
+
+    const groupBySection = (items: { item: Item, quantity: string }[]): Record<string, {
+        name: string,
+        quantity: string
+    }[]> => {
+        return items.reduce((acc, item) => {
+            if (!acc[item.item.section]) {
+                acc[item.item.section] = [];
+            }
+            acc[item.item.section].push({
+                name: item.item.name,
+                quantity: item.quantity
+            });
+            return acc;
+        }, {} as Record<string, { name: string, quantity: string }[]>);
+    };
+
+    const groupedItems = groupBySection(items);
+
+    function capitalizeFirstLetter(str: string): string {
+        return str.charAt(0).toUpperCase() + str.slice(1);
+    }
+
+    return (
+        <>
+            <h3>Items:</h3>
+            <div>
+                {Object.keys(groupedItems).map((section) => (
+                    <div key={section}>
+                        <h2>{capitalizeFirstLetter(section)}:</h2>
+                        <ul>
+                            {groupedItems[section].map((item, index) => {
+                                if (checkbox) {
+                                    return <ListItemWithCheckbox key={index} item={item} />
+                            } else {
+                                return <ListItemWithoutCheckbox key={index} item={item}/>}
+                            })}
```

Comment:
Usually devs use the ternary operator to conditionally render components, so that there aren't multiple `return` statements in the code, which can be confusing. 

```suggestion
{groupedItems[section].map((item) =>
    checkbox ? (
        <ListItemWithCheckbox key={item.name} item={item} />
    ) : (
        <ListItemWithoutCheckbox key={item.name} item={item} />
    )
)}
```